src/types/options/qualityRequestOptions.ts

```diff
@@ -0,0 +1,102 @@
+import { CoreServiceUrl, OSMType, PoiGroup, TravelMode, TravelTimeFactors } from '..';
+
+export interface QualityRequestOptions {
+  /** Criterion definitions
+   * For each criterion, a key must be set to be able to identify the different criteria in the response */
+  [criteriaId: string]: (PointOfInterestCriterion | StatisticsCriterion | GravitationalCriterion)
+}
+
+
+interface BaseCriterion {
+  /** This is the service URL to which the Targomo Reachability requests are dispatched. The endpoint has to correspond to the addresses
+   * @example if you want to execute reachabilities in Belgium you need to point to a Targomo Endpoint
+   * that includes Belgium geographically (https://api.targomo.com/westcentraleurope/) */
+  coreServiceUrl: CoreServiceUrl
+
+  /** Determines the dimension of the edges' weights, i.e. time (distance in seconds) or distance (distance in meters).
+   * 'distance' Will optimize for distance and search for the shortest route,
+   * 'time' will optimize for time and will search for the fastest route.
+   * Distance cannot be used with travelType 'transit' and with Gravitational Criterion */
+  edgeWeight: ('time' | 'distance')
+
+  /** The maximum distance 'depth' of the built network in seconds (for edgeWeight = time) or meters (for edgeWeight = distance).
+   * The upper limit of this variable is based on your subscription plan and differs between distance and time.
+If it is set too low routes between points won't be found */
+  maxEdgeWeight: number
+
+  /** This field defines what travel mode is used for routing.
+   * All travel modes are build similarly but differ in some way, 'rush-hour' for 'car' or the 'date/time' for 'transit'. */
+  travelMode: TravelMode
+
+  /** Whether or not the elevation heuristic will be used (e.g. downhill with bike quicker than uphill).
+   * @default true
+   */
+  elevation?: boolean
+
+  travelTimeFactors?: TravelTimeFactors
+}
+
+
+export interface PointOfInterestCriterion extends BaseCriterion {
+  /** must be one of these values
+  @example poiCoverageCount: the retrieved score is the number of POIs reachable from the location
+  @example poiCoverageDistance: the retrieved score is the sum of the scores weighted by the distance of each reachable POI.
+   * The higher the distance between the location and the considered POI is, the smaller is the score.
+  @example closestPoiDistance: the retrieved score is the score weighted by the distance of the closest POI reachable from the location
+    */
+  type: ('closestPoiDistance' | 'poiCoverageCount' | 'poiCoverageDistance')
+  /** list of Osm Types to consider for this criterion */
+  osmTypes: (OSMType[] | PoiGroup[])
```

Comment:
you can actually combine OSMType and POIGroup is the same array
So I would probably put more `(OSMType | PoiGroup)[]`. I don't know if it is valid in typescript though.

Reply:
Oh right, I misread that info. That syntax is valid.